File: FournilBack/src/main/java/fr/fournil/bakery/model/entities/DeliveryPointAdressAndDates.java
package fr.fournil.bakery.model.entities;




import java.time.DayOfWeek;
import java.util.HashSet;
import java.util.Set;

import javax.persistence.CascadeType;
import javax.persistence.Column;
import javax.persistence.Convert;
import javax.persistence.Entity;
import javax.persistence.FetchType;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.OneToMany;
import javax.persistence.OneToOne;
import javax.persistence.Table;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;

import org.hibernate.annotations.NaturalId;

@Entity
@Table(name="deliveryPointAdressAndDates")
public class DeliveryPointAdressAndDates {
	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long id;
	
	//@NaturalId
   // @Column(columnDefinition = "TEXT")
	private String deliveryPointAdress;
	
	//@NaturalId
	//@Column(name = "deliveryDate",columnDefinition="int", insertable=false)
	//@Temporal(TemporalType.TIMESTAMP) //(columnDefinition = "TEXT")
    private  DayOfWeek deliveryDate;
	
	//@NaturalId
	//@Column(name = "orderDeadline",columnDefinition="int", insertable=false)
	//@Temporal(TemporalType.TIMESTAMP) //(columnDefinition = "TEXT") //(columnDefinition = "TEXT")
	
	private DayOfWeek orderDeadline;
    
	
	
	@OneToMany(mappedBy = "deliveryPointAdressAndDates", cascade = {CascadeType.PERSIST}, fetch = FetchType.EAGER)
	private Set<Commande> commandeSet = new HashSet<Commande>();

	public DeliveryPointAdressAndDates() {
		super();
	}

	
	public DeliveryPointAdressAndDates(Long id, String deliveryPointAdress, DayOfWeek deliveryDate,
			DayOfWeek orderDeadline, Set<Commande> commandeSet) {
		super();
		this.id = id;
		this.deliveryPointAdress = deliveryPointAdress;
		this.deliveryDate = deliveryDate;
		this.orderDeadline = orderDeadline;
		this.commandeSet = commandeSet;
	}


	public DeliveryPointAdressAndDates(String deliveryPointAdress, DayOfWeek deliveryDate, DayOfWeek orderDeadline,
			Set<Commande> commandeSet) {
		super();
		this.deliveryPointAdress = deliveryPointAdress;
		this.deliveryDate = deliveryDate;
		this.orderDeadline = orderDeadline;
		this.commandeSet = commandeSet;
	}


	public Long getId() {
		return id;
	}


	public void setId(Long id) {
		this.id = id;
	}


	public String getDeliveryPointAdress() {
		return deliveryPointAdress;
	}


	public void setDeliveryPointAdress(String deliveryPointAdress) {
		this.deliveryPointAdress = deliveryPointAdress;
	}


	public DayOfWeek getDeliveryDate() {
		return deliveryDate;
	}


	public void setDeliveryDate(DayOfWeek deliveryDate) {
		this.deliveryDate = deliveryDate;
	}


	public DayOfWeek getOrderDeadline() {
		return orderDeadline;
	}


	public void setOrderDeadline(DayOfWeek orderDeadline) {
		this.orderDeadline = orderDeadline;
	}


	public Set<Commande> getCommandeSet() {
		return commandeSet;
	}


	public void setCommandeSet(Set<Commande> commandeSet) {
		this.commandeSet = commandeSet;
	}


	@Override
	public String toString() {
		return "DeliveryPointAdressAndDates [id=" + id + ", deliveryPointAdress=" + deliveryPointAdress
				+ ", deliveryDate=" + deliveryDate + ", orderDeadline=" + orderDeadline
				+ "]";
	}


	

	

	

	
	

	
	
}

File: FournilFront/src/app/models/delivery-point-adress-and-dates.ts
import { Commande } from './commande';

export class DeliveryPointAdressAndDates {

    private _id: number;
    private _deliveryPointAdress: String;
    private _deliveryDate: Date;
    private _orderDeadline: Date;
    private _commandeSet: Commande[];


	constructor(id: number, deliveryPointAdress: String, deliveryDate: Date, orderDeadline: Date, commandeSet: Commande[]) {
		this._id = id;
		this._deliveryPointAdress = deliveryPointAdress;
		this._deliveryDate = deliveryDate;
		this._orderDeadline = orderDeadline;
		this._commandeSet = commandeSet;
	}

    /**
     * Getter id
     * @return {number}
     */
	public get id(): number {
		return this._id;
	}

    /**
     * Getter deliveryPointAdress
     * @return {String}
     */
	public get deliveryPointAdress(): String {
		return this._deliveryPointAdress;
	}

    /**
     * Getter deliveryDate
     * @return {Date}
     */
	public get deliveryDate(): Date {
		return this._deliveryDate;
	}

    /**
     * Getter orderDeadline
     * @return {Date}
     */
	public get orderDeadline(): Date {
		return this._orderDeadline;
	}

    /**
     * Getter commandeSet
     * @return {Commande[]}
     */
	public get commandeSet(): Commande[] {
		return this._commandeSet;
	}

    /**
     * Setter id
     * @param {number} value
     */
	public set id(value: number) {
		this._id = value;
	}

    /**
     * Setter deliveryPointAdress
     * @param {String} value
     */
	public set deliveryPointAdress(value: String) {
		this._deliveryPointAdress = value;
	}

    /**
     * Setter deliveryDate
     * @param {Date} value
     */
	public set deliveryDate(value: Date) {
		this._deliveryDate = value;
	}

    /**
     * Setter orderDeadline
     * @param {Date} value
     */
	public set orderDeadline(value: Date) {
		this._orderDeadline = value;
	}

    /**
     * Setter commandeSet
     * @param {Commande[]} value
     */
	public set commandeSet(value: Commande[]) {
		this._commandeSet = value;
	}



}